mantidproject/mantid
# IDA Convfit get lorentzian width from ipf

**Description of work.**

This PR makes it so that the Convfit tab in the indirct data analysis gui can get it's initial guess paramiters for it's lorenzian from the resolution IPF when it is loaded. 
Also as part of finding a solution to this I have also added a propper overload for IFunction getError and SetError by paramiter name instead of just index.

**To test:**

1. get data from #24664 
1. Open workbench
2. Open Indirect->Data Analysis
3. Go to Convfit
4. In the function browser add a lorenzian
5. Load a sample resolution pair
6. The function browser shoud have it's FWHM changed to 0.0175, the rage markers on the plor should have been updated to reflect this.
7. change to multiple input and do the same with indevidual spectra and combinations of different workspaces.

Fixes #29254 

The changes to convfit has release notes, but the overload for getError and setError are internal only and do not require them.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
